adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java

```diff
@@ -83,7 +87,9 @@ public static MqttContext fromPublishPacket(
         result.authenticatedDevice = authenticatedDevice;
         if (publishedMessage.topicName() != null) {
             try {
-                result.topic = ResourceIdentifier.fromString(publishedMessage.topicName());
+                final PropertyBag propertyBag = PropertyBag.fromTopic(publishedMessage.topicName());
+                result.timeToLive = extractTimeToLive(propertyBag);                
```

Comment:
IMHO the context should provide access to the PropertyBag itself but it should have no knowledge about the semantics of particular properties contained in it, don't you think?

Reply:
@sophokles73 IMHO `PropertyBag` class should not know the semantics of the properties. It should be straight forward knowing how to extract the properties and retrieve a particular property without any context.  With that in my mind, the next suitable place for the `time-to-live` is `MQTTContext` as it already holds information required to process a message such as topic, message, endpoint etc.

Reply:
No, PropertyBag should not know about *time-to-live* but MqttContext shouldn't either. My point is that the MQTT adapter should do something like
```
context.getPropertyBag().getProperty("ttl");
```
Otherwise, I would wonder why the MqttContext knows about ttl but not about the other properties contained in the PropertyBag? How does the MQTT adapter get access to these?

Reply:
I also tempted to the add the `propertyBag` as an attribute in `MQTTContext`.  Right now we use the `property-bag` only for `time-to-live` and there are no other use cases. Hence I exposed only `time-to-live` in `MQTTContext` keeping in mind that in future if any use case pops up, it would be still easy to expose it as it is already available in `MQTTContext`. It is also fine for me to provide already access to the `propertyBag` and do the computation of `time-to-live` in the _MQTTAdapter_ base class and also making it reusable by the custom MQTT adapters.

Reply:
that would be great :+1:

Reply:
@sophokles73 I have pushed the changes. Please have a look.